packages/redis/src/RedisClient.ts

```diff
@@ -20,24 +20,60 @@ import {
 } from "@walmartlabs/cookie-cutter-core";
 import { Span, SpanContext, Tags, Tracer } from "opentracing";
 import { isString } from "util";
-import { IRedisOptions, IRedisClient, RedisStreamID } from ".";
-import { RedisProxy } from "./RedisProxy";
+import { IRedisOptions, IRedisClient, AutoGenerateRedisStreamID } from ".";
+import { RedisProxy, StreamResult } from "./RedisProxy";
 
 enum RedisMetrics {
     Get = "cookie_cutter.redis_client.get",
     Set = "cookie_cutter.redis_client.set",
     XAdd = "cookie_cutter.redis_client.xadd",
+    XRead = "cookie_cutter.redis_client.xread",
+    XReadGroup = "cookie_cutter.redis_client.xreadgroup",
+    XGroupCreate = "cookie_cutter.redis_client.xgroup.create",
+    XGroupAlreadyExists = "cookie_cutter.redis_client.xgroup.already_exists",
+    XAck = "cookie_cutter.redis_client.xack",
+    xPending = "cookie_cutter.redis_client.xpending",
+    xClaim = "cookie_cutter.redis_client.xclaim",
 }
 
 enum RedisMetricResults {
     Success = "success",
     Error = "error",
 }
 
-enum RedisOpenTracingTagKeys {
+export enum RedisOpenTracingTagKeys {
     BucketName = "redis.bucket",
 }
 
+function extractStreamValue(results: StreamResult): string {
+    // Since our initial implementation on pulls 1 value from 1 stream at a time
+    // there should only 1 item in results
```

Comment:
nit: small typo
```suggestion
    // Since our initial implementation only pulls 1 value from 1 stream at a time
    // there should only be 1 item in results
```